src/workload.rs

```diff
@@ -387,13 +402,15 @@ impl WorkloadInformation {
 
 /// A WorkloadStore encapsulates all information about workloads in the mesh
 #[derive(serde::Serialize, Default, Debug)]
-pub struct WorkloadStore {
+pub struct WorkloadStore<'a> {
     workloads: HashMap<IpAddr, Workload>,
     // workload_to_vip maintains a mapping of workload IP to VIP. This is used only to handle removals.
     workload_to_vip: HashMap<IpAddr, HashSet<(SocketAddr, u16)>>,
     // vips maintains a mapping of socket address with service port to workload ip and socket address
     // with target ports in hashset.
     vips: HashMap<SocketAddr, HashSet<(IpAddr, u16)>>,
+    // for fetching certificates on workload discovery
+    prefetch: &'a dyn Fn(&Identity) -> dyn Future<Output=Result<Certs, Error>>
```

Comment:
might consider wrapping this in an Arc so we don't need all the lifetimes.

Reply:
And pass a `Box<dyn CertificateProvider>` directly instead of a closure will simplify. Note we already do this in other places